### PR TITLE
Improve Benchmark Selection

### PR DESCRIPTION
**Pull request by willtebbutt**
@yebai pointed out that the benchmarks give a somewhat pessimistic lens into Tapir.jl's performance. This PR will play around with some of the parameters of the benchmarks to see if a more balanced view can be obtained.

### CODE REVIEW

**Comment by willtebbutt** [comment]
Correct. I should put this on the post!

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -122,6 +122,6 @@ jobs:
         uses: peter-evans/create-or-update-comment@v4
         with:
           issue-number: ${{ github.event.pull_request.number }}
-          body: "Performance Ratio:\nWarning: results are very approximate!\n```\n${{ steps.read-file.outputs.table }}\n```"
+          body: "Performance Ratio:\nRatio of time to compute gradient and time to compute function.\nWarning: results are very approximate! See bench/README.md for more context.\n```\n${{ steps.read-file.outputs.table }}\n```"
```
Comment (yebai) [inline .github/workflows/CI.yml]: Consider making it a clickable link?

File: bench/README.md [hunk 2]
```diff
@@ -43,6 +43,12 @@ plot_ratio_histogram!(df)
 ## Inter-framework Benchmarking
 
 This comprises a small suite of functions that we AD using `Tapir.jl`, `Zygote.jl`, `ReverseDiff.jl`, and `Enzyme.jl`.
+The primary purpose of this suite of benchmarks is to ensure that we're regularly comparing the performance of a range of reverse-mode ADs on a set of problems which are known to stretch them in various ways.
```
Comment (yebai) [inline bench/README.md:46]: Consider adding some comments on why Zygote is fast (~~due to more rules~~; you already added it below), and relevant discussions on other AD we had previously?

File: bench/README.md [hunk 3]
```diff
@@ -43,6 +43,12 @@ plot_ratio_histogram!(df)
...
+For any given function in the suite, some frameworks might have rules for it, and some not.
+Consequently, they're not comparing the same thing in all cases.
+
+Please note that we have found that the results of the comparisons vary widely from machine to machine.
```
Comment (yebai) [inline bench/README.md]: Not sure this is generally true; the sum example seems to have a big variance, but other numbers are relatively stable across machines.

File: .github/workflows/CI.yml (hunk 1 above)
Comment (willtebbutt) [reply .github/workflows/CI.yml]: Good idea

File: bench/README.md (hunk 3 above)
Comment (willtebbutt) [reply bench/README.md]: This is probable accurate I've just removed this comment entirely.

File: bench/README.md (hunk 2 above)
Comment (willtebbutt) [reply bench/README.md:46]: I've modified this to be a bit more precise.

**Comment by willtebbutt** [comment]
@yebai @sunxd3 thanks for the feedback on this. If you're happy with it, I'll merge when CI passes.

File: .github/workflows/CI.yml [hunk 4]
```diff
@@ -122,6 +122,6 @@ jobs:
         uses: peter-evans/create-or-update-comment@v4
         with:
           issue-number: ${{ github.event.pull_request.number }}
-          body: "Performance Ratio:\nRatio of time to compute gradient and time to compute function.\nWarning: results are very approximate! See bench/README.md for more context.\n```\n${{ steps.read-file.outputs.table }}\n```"
+          body: "Performance Ratio:\nRatio of time to compute gradient and time to compute function.\nWarning: results are very approximate! See https://github.com/compintell/Tapir.jl/tree/main/bench for more context.\n```\n${{ steps.read-file.outputs.table }}\n```"
```
Comment (yebai) [inline .github/workflows/CI.yml]:
```suggestion
          body: "Performance Ratio:\nRatio of time to compute gradient and time to compute function.\nWarning: results are very approximate! See [here](https://github.com/compintell/Tapir.jl/tree/main/bench#inter-framework-benchmarking) for more context.\n```\n${{ steps.read-file.outputs.table }}\n```"
```

**Comment by willtebbutt** [comment]
This all seems reasonable -- they'll be easiest to write once the docs are further along, and I feel like they ought to live in the docs anyway.

**Pull request merged**